code/obj/machinery/computer/chem_requester.dm

```diff
@@ -0,0 +1,152 @@
+var/list/datum/chem_request/chem_requests = list()
+
+/datum/chem_request
+	var/requester_name = ""
+	var/reagent_name = ""
+	var/reagent_color = null
+	var/note = ""
+	var/volume = 5
+	var/area_name = "Somewhere"
+	var/state = "pending"
+	var/id
+	var/static/last_id = 0
+	// the tick count this request was placed on
+	var/time = 0
+	New()
+		..()
+		src.id = ++last_id
+
+/obj/machinery/computer/chem_requester
+	name = "Chemical request console"
+	icon = 'icons/obj/computer.dmi'
+	icon_state = "chemreq"
+	var/datum/chem_request/request = new
+	var/obj/item/card/id/card = null
+	var/max_volume = 400
+	var/area_name = null
+
+	ui_data(mob/user)
+		. = list()
+		if (src.card)
+			.["card"] = list("name" = src.card.registered, "role" = src.card.assignment)
+		else
+			.["card"] = null
+		.["selected_reagent"] = src.request.reagent_name
+		.["notes"] = src.request.note
+		.["volume"] = src.request.volume
+
+	ui_static_data(mob/user)
+		. = list()
+		var/list/chems = list()
+		for (var/id in chem_reactions_by_id)
+			var/datum/chemical_reaction/reaction = chem_reactions_by_id[id]
+			if (reaction.result && !reaction.hidden)
+				var/datum/reagent/reagent = reagents_cache[reaction.result]
+				if (reagent && !istype(reagent, /datum/reagent/fooddrink)) //all the cocktails clog the UI
+					chems[lowertext(reagent.name)] = reagent.id
+		.["chemicals"] = sortList(chems)
+		.["max_volume"] = src.max_volume
+
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if (!ui)
+			ui = new(user, src, "ChemRequester")
+			ui.open()
+
+	ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+		switch (action)
+			if ("reset_id")
+				src.card = null
+				qdel(src.request)
+				src.request = new
+				. = TRUE
+			if ("set_reagent")
+				src.request.reagent_name = params["reagent_name"]
+				var/datum/reagent/reagent = reagents_cache[params["reagent_id"]]
+				if (reagent)
+					src.request.reagent_color = list(reagent.fluid_r, reagent.fluid_g, reagent.fluid_b)
+				. = TRUE
+			if ("set_notes")
+				src.request.note = strip_html(copytext(params["notes"], 1, 66))
+			if ("set_volume")
+				src.request.volume = clamp(params["volume"], 1, src.max_volume)
+				. = TRUE
+			if ("submit")
+				src.request.area_name = src.area_name || (get_area(src))?.name || src.request.area_name
+				src.request.requester_name = src.card.registered + " ([src.card.assignment])"
+				src.request.time = ticker.round_elapsed_ticks
+				//byond jank, lists are only associative if they aren't int indexed
+				chem_requests["[src.request.id]"] = src.request
```

Comment:
might be cooler to store this on the mainframe or something, packets so nerds can sniff and see?
(not required, idea)